TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/ArmSystem.java

```diff
@@ -105,343 +87,147 @@ private static Direction reverse(Direction direction){
-        slider.setPower(0);
-        /*
-        // At rest, pre-int, this will be false, i.e. the limit switch IS pressed.
-        if (limitSwitch.getState()) {
-            // Limit switch has been released, cord is under tension.
-            // With cord under tension, lower until switch is depressed.
-            direction = Direction.DOWN;
-        }
-
-        Log.d(TAG,"Calibrating");
-        Log.d(TAG, "Direction: " + direction);
-        Log.d(TAG, "Switch State: " + limitSwitch.getState());
-        // If we're going down and we hit the switch, then we're done
-        if (direction == Direction.DOWN && !limitSwitch.getState()) {
-            slider.setPower(0);
-            calibrated = true;
-            calibrationDistance = slider.getCurrentPosition();
-            slider.setDirection(Direction.motorDirection(Direction.UP));
-            setSliderHeight(0);
-        }
-
-         */
-    }
-
-    private void stop() {
-        slider.setPower(0);
+    private void movePresetPosition(Position pos){
+        double[] posArray = pos.getPos();
+        servoEnumMap.get(ServoNames.PIVOT).setPosition(posArray[0]);
+        servoEnumMap.get(ServoNames.ELBOW).setPosition(posArray[1]);
+        servoEnumMap.get(ServoNames.WRIST).setPosition(posArray[2]);
     }
 
     // Pos should be the # of blocks high it should be
-    private void setSliderHeight(int pos) {
-        targetHeight = pos;
-        slider.setTargetPosition(calculateHeight(targetHeight));
-        slider.setDirection(Direction.motorDirection(Direction.UP));
+    public void setSliderHeight(double pos) {
+        mTargetHeight = Range.clip(pos, 0, 5);
+        setPosTarget();
         slider.setMode(DcMotor.RunMode.RUN_TO_POSITION);
-        Log.d(TAG, "Set target height to" + calculateHeight(targetHeight));
+    }
+
+    public void setSliderHeight(int pos) {
+        // * 1.0 converts to double
+        setSliderHeight(pos * 1.0);
     }
 
     // Little helper method for setSliderHeight
-    private int calculateHeight(int pos) {
-        return START_HEIGHT + calibrationDistance + (pos * INCREMENT_HEIGHT);
+    private int calculateHeight(double pos){
+        return (int) (pos == 0 ? mCalibrationDistance - 20 : mCalibrationDistance + (pos * INCREMENT_HEIGHT));
     }
 
-    // Should be called every loop
-    private void updateHeight(double speed) {
-        slider.setPower(speed);
-        slider.setTargetPosition(calculateHeight(targetHeight));
-        if (!fastMode) {
-            updateServo(elbow, elbowTarget);
-            toReturn += elbowTarget + "\n";
-            updateServo(pivot, pivotTarget);
-            toReturn += pivotTarget + "\n";
-            updateServo(wrist, wristTarget);
-            toReturn += wristTarget + "\n";
+    // Must be called every loop
+    public boolean runSliderToTarget(double speed){
+        int targetHeight = calculateHeight(mTargetHeight);
+        if (slider.getCurrentPosition() != targetHeight) {
```

Comment:
Isn't `targetHeight` equal to `slider.getTargetPosition()`? We don't need to recalculate it here.